pkg/apis/extensions/validation/validation.go

```diff
@@ -741,3 +741,52 @@ func ValidateNetworkPolicyUpdate(update, old *extensions.NetworkPolicy) field.Er
 	}
 	return allErrs
 }
+
+// ValidateStorageClass validates a StorageClass.
+func ValidateStorageClass(storageClass *extensions.StorageClass) field.ErrorList {
+	allErrs := apivalidation.ValidateObjectMeta(&storageClass.ObjectMeta, false, apivalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
+	allErrs = append(allErrs, validateProvisionerType(storageClass.ProvisionerType, field.NewPath("provisionerType"))...)
+	allErrs = append(allErrs, validateProvisionerParameters(storageClass.ProvisionerParameters, field.NewPath("provisionerParameters"))...)
+
+	return allErrs
+}
+
+// ValidateStorageClassUpdate tests if required fields in the StorageClass are set.
+func ValidateStorageClassUpdate(storageClass, oldStorageClass *extensions.StorageClass) field.ErrorList {
```

Comment:
Are we allowing updates?  Do we fully comprehend the implications of changing a class while existing volumes might reference it?  Where is that documented (hint: it should be in the API spec).

If we're not ready for that, we can disallow updates until we know what we want to do with it.  This is a strategy that has been used well and probably not often enough.